e2e/test/iothub/method/MethodE2ETests.cs

```diff
@@ -466,9 +468,22 @@ public static async Task<Task> SetModuleReceiveMethodAsync(IotHubModuleClient mo
                 responseTimeout,
                 serviceClientTransportSettings);
 
-            await Task.WhenAll(serviceSendTask, methodReceivedTask).ConfigureAwait(false);
+            Task.WaitAll(
+                new Task[]
+                {
+                    serviceSendTask,
+                    methodReceivedTask
+                },
+                TimeSpan.FromMinutes(1));
 
             await deviceClient.CloseAsync().ConfigureAwait(false);
+
+            // If the Task.WaitAll timed out, then one or both of these tasks timed out. This block clarifies which task(s) timed out
+            using (new AssertionScope())
```

Comment:
Here's a neat trick I picked up. In times like these where you have multiple assertions to make, blocks like this allow you to throw for **all** of the assertions that failed instead of just the **first** assertion that failed.